### PR TITLE
tests: update tests

### PR DESCRIPTION
**Pull request by Kamoppl**

### CODE REVIEW

File: test/functional/tests/cache_ops/test_seq_cutoff.py [hunk 1]
```diff
@@ -26,187 +25,260 @@ class VerifyType(Enum):
...
+
+    with TestRun.step(
+        f"Start cache in {cache_mode} mode and add {len(core_parts)} cores to the cache"
+    ):
+        cache = casadm.start_cache(
+            cache_dev=cache_part, cache_mode=cache_mode, force=True, cache_line_size=cache_line_size
+        )
+        core_list = [cache.add_core(core_dev=core_part) for core_part in core_parts]
+
+    with TestRun.step(f"Set sequential cut-off parameters for all cores"):
```
Comment (Deixx) [inline test/functional/tests/cache_ops/test_seq_cutoff.py:77]:
```suggestion
    with TestRun.step("Set sequential cut-off parameters for all cores"):
```

File: test/functional/tests/basic/test_basic.py [hunk 2]
```diff
@@ -69,9 +68,11 @@ def test_negative_start_cache():
       - Fails to start cache on another partition with the same id
     """
 
-    with TestRun.step("Partition cache devices"):
+    with TestRun.step("Prepare cache and core devices"):
```
Comment (Deixx) [inline test/functional/tests/basic/test_basic.py:71]: what core?

File: test/functional/tests/cache_ops/test_concurrent_flushes.py [hunk 3]
```diff
@@ -1,195 +1,215 @@
...
-                .count(count) \
-                .run()
+            fio = (
+                Fio()
+                .create_command()
+                .target(core)
+                .io_engine(IoEngine.libaio)
+                .block_size(Size(4, Unit.MebiByte))
+                .size(core.size)
+                .read_write(ReadWrite.randwrite)
```
[the remark this answers is not in the document]
Comment (Deixx) [reply test/functional/tests/cache_ops/test_concurrent_flushes.py:194]: you fixed other occurrence, fix this too ;)

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 4]
```diff
@@ -28,6 +29,119 @@
...
+                TestRun.fail(f"Core {core.core_id} should be active but is {core.get_status()}.")
+
+    with TestRun.step("Stop cache"):
+        cache.stop()
+
+    with TestRun.step("Add cores to core pool"):
+        core_pool_list = [casadm.try_add(core_device=core_dev, cache_id=1, core_id=core_id)
+                          for core_dev, core_id in zip(core_dev_list, range(1, core_count))]
+
+    with TestRun.step("Check if all cores in core pool has detached status"):
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:73]: have

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 5]
```diff
@@ -28,6 +29,119 @@
...
+                    f"Core {core.core_id} should be active but is {core.get_status()}."
+                    )
+
+        hidden_core = cache.get_core_devices()[2]
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Hide last core partition"):
+        core_device.remove_partition(core_device.partitions[-1])
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:130]: decide on one concept for the last element of the list - either [2] or [-1]

File: test/functional/tests/initialize/test_recovery.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(2, Unit.GibiByte)])
+        core_device.create_partitions([Size(4, Unit.GibiByte)])
+
+        cache_dev = cache_device.partitions[0]
+        core_dev = core_device.partitions[0]
+
+    with TestRun.step("Start caches and add cores"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:52]: core

File: test/functional/tests/initialize/test_recovery.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+        cache = casadm.start_cache(cache_dev=cache_dev, cache_mode=CacheMode.WB)
+        core = cache.add_core(core_dev=core_dev)
+
+    with TestRun.step("Create filesystem on core device"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:56]: core (w/o device)

File: test/functional/tests/initialize/test_recovery.py [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+        core.create_filesystem(filesystem)
+
+    with TestRun.step("Mount cache device"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:59]: core

File: test/functional/tests/initialize/test_recovery.py [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            Dd()
+            .input("/dev/urandom")
+            .output(f"{mount_point}/test")
+            .count(1)
+            .block_size(Size(50, Unit.MegaByte))
+        )
+        dd.run()
+        sync()
+
+    with TestRun.step("Calculate cache md5sums before unplug"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:73]: core

File: test/functional/tests/initialize/test_recovery.py [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+        core_mnt_md5s_before = File(f"{mount_point}/test").md5sum()
+
+    with TestRun.step("Unmount core device"):
+        core.unmount()
+
+    with TestRun.step("Turn off cache device"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:79]: Unplug

File: test/functional/tests/initialize/test_recovery.py [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+        cache_device.unplug()
+
+    with TestRun.step("Stop caches without flushing and check error message"):
+        output = TestRun.executor.run(stop_cmd(cache_id=str(cache.cache_id), no_data_flush=True))
+        if len(get_caches()) > 0:
+            TestRun.fail("CAS failed to stop cache")
+        if not check_stderr_msg(output, stop_cache_errors):
+            TestRun.fail(f"Wrong error message during cache stop")
+
+    with TestRun.step("Turn on devices."):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:89]: Plug cache device

File: test/functional/tests/initialize/test_recovery.py [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+    with TestRun.step("Stop cache"):
+        cache.stop()
+
+    with TestRun.step("Mount core device"):
+        mount(core.core_device, mount_point)
+        core.mount_point = mount_point
+        if not check_if_file_exists(core.mount_point):
+            TestRun.LOGGER.error(f"Mounting core device {mount_point} failed")
+
+    with TestRun.step("Calculate cache md5sums after recovery"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:104]: core device

File: test/functional/tests/initialize/test_recovery.py [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+        core_mnt_md5s_after = File(f"{core.mount_point}/test").md5sum()
+
+    with TestRun.step("Compare md5 sums for cores and core devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:107]: core and core device

File: test/functional/tests/initialize/test_recovery.py [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(2, Unit.GibiByte)])
+        core_device.create_partitions([Size(4, Unit.GibiByte)])
+
+        cache_dev = cache_device.partitions[0]
+        core_dev = core_device.partitions[0]
+
+    with TestRun.step("Start caches and add cores"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:140]: as above

File: test/functional/tests/initialize/test_recovery.py [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+        cache = casadm.start_cache(cache_dev=cache_dev, cache_mode=CacheMode.WB)
+        core = cache.add_core(core_dev=core_dev)
+
+    with TestRun.step("Create filesystem on core devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:144]: as above

File: test/functional/tests/initialize/test_recovery.py [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+        core.create_filesystem(filesystem)
+
+    with TestRun.step("Mount cache devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:147]: as above

File: test/functional/tests/initialize/test_recovery.py [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+            Dd()
+            .input("/dev/urandom")
+            .output(f"{mount_point}/test")
+            .count(1)
+            .block_size(Size(50, Unit.MegaByte))
+        )
+        dd.run()
+        sync()
+
+    with TestRun.step("Calculate cache md5sums before unplug"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:161]: as above

File: test/functional/tests/initialize/test_recovery.py [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+        core_mnt_md5s_before = File(f"{mount_point}/test").md5sum()
+
+    with TestRun.step("Umount core devices"):
+        core.unmount()
+
+    with TestRun.step("Turn off cache devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:167]: device

File: test/functional/tests/initialize/test_recovery.py [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+        cache_device.unplug()
+
+    with TestRun.step("Stop caches without flushing and check error message"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:170]: cache

File: test/functional/tests/initialize/test_recovery.py [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+        output = TestRun.executor.run(stop_cmd(cache_id=str(cache.cache_id), no_data_flush=True))
+        if len(get_caches()) > 0:
+            TestRun.fail("CAS failed to stop cache")
+        if not check_stderr_msg(output, stop_cache_errors):
+            TestRun.fail(f"Wrong error message during cache stop")
+
+    with TestRun.step("Turn on devices."):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:177]: device

File: test/functional/tests/initialize/test_recovery.py [hunk 21]
```diff
@@ -0,0 +1,196 @@
...
+        cache_device.plug()
+
+    with TestRun.step("Load caches with metadata"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:180]: just "Load cache" (cache cannot be loaded without metadata)

File: test/functional/tests/initialize/test_recovery.py [hunk 22]
```diff
@@ -0,0 +1,196 @@
...
+        cache = casadm.load_cache(cache_dev)
+
+    with TestRun.step("Mount exported object"):
+        core.mount(mount_point)
+        if not check_if_file_exists(core.mount_point):
+            TestRun.LOGGER.error(f"Mounting core device {mount_point} failed")
+
+    with TestRun.step("Calculate cache md5sums after recovery"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:188]: test file

File: test/functional/tests/initialize/test_recovery.py [hunk 23]
```diff
@@ -0,0 +1,196 @@
...
+        core_mnt_md5s_after = File(f"{core.mount_point}/test").md5sum()
+
+    with TestRun.step("Compare md5 sums for cores and core devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:191]: Compare test file md5sums

File: test/functional/tests/initialize/test_simulation_startup.py [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+def test_simulation_startup_from_config():
+    """
+    title: Test for CAS initialization from a configuration file simulated during boot
+    description: |
+        Verify that CAS can be properly initialized from a configuration file and subsequently
+        started correctly during a simulated system boot process.
+    pass_criteria:
+      - Cache initialization from configuration file works properly
+      - Cache is working after simulated boot
+      - Core is working after simulated boot
```
Comment (Deixx) [inline test/functional/tests/initialize/test_simulation_startup.py:32]: I wouldn't call triggering udev a booting simulation ;)

File: test/functional/tests/initialize/test_simulation_startup.py [hunk 25]
```diff
@@ -0,0 +1,84 @@
...
+    with TestRun.step("Verify if core is working"):
+        core = cache.get_core_devices()[0]
+        if core.get_status() is not CoreStatus.active:
+            TestRun.fail(f"Core {core.core_id} should be active but is in {core.get_status()} "
+                         f"state.")
+
+    with TestRun.step("Stop cache instance"):
+        TestRun.executor.run_expect_success(ctl_stop())
+
+    with TestRun.step("Simulate booting process"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_simulation_startup.py:71]: as above

File: test/functional/tests/initialize/test_simulation_startup.py [hunk 26]
```diff
@@ -0,0 +1,84 @@
...
+            TestRun.fail(f"Cache {cache.cache_id} should be running but is in {cache.get_status()} "
+                         f"state.")
+
+    with TestRun.step("Verify if core is working"):
+        core = cache.get_core_devices()[0]
+        if core.get_status() is not CoreStatus.active:
+            TestRun.fail(f"Core {core.core_id} should be active but is in {core.get_status()} "
+                         f"state.")
+
+    with TestRun.step("Stop cache instance"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_simulation_startup.py:68]: in this case specify that the cache was stopped using casctl

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 27]
```diff
@@ -0,0 +1,129 @@
...
+            if cache.get_cleaning_policy() != policy_to_change:
+                TestRun.fail("Wrong cleaning policy after chane it during I/O")
+
+
+@pytest.mark.parametrizex("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_change_cleaning_policy_during_io_fs(filesystem):
+    """
+    title: Test for changing the cleaning policy during IO.
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:79]: titles should differentiate tests - add info about filesystem or its absence to at least one of these tests

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 28]
```diff
@@ -0,0 +1,129 @@
...
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_change_cleaning_policy_during_io_raw():
+    """
+    title: Test for changing the cleaning policy during background IO.
+    description: |
+        Stress test to change the cleaning policy during background IO operations on CAS raw
+        device with a Write-Back cache mode.
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:31]: 'raw exported object' or 'raw cached volume'

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+        )
+
+        fio_pid = fio.run_in_background()
+
+    with TestRun.step(f"Start changing the cleaning policy during I/O operations"):
+        while TestRun.executor.check_if_process_exists(fio_pid):
+            policy_to_change = random.choice(list(CleaningPolicy))
+            cache.set_cleaning_policy(policy_to_change)
+            if cache.get_cleaning_policy() != policy_to_change:
+                TestRun.fail("Wrong cleaning policy after chane it during I/O")
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:71]: changing it

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+            .read_write(ReadWrite.randrw)
+            .io_engine(IoEngine.sync)
+            .block_size(Size(1, Unit.Blocks4096))
+        )
+
+        fio_pid = fio.run_in_background()
+
+    with TestRun.step(f"Start changing the cleaning policy during I/O operations"):
+        while TestRun.executor.check_if_process_exists(fio_pid):
+            policy_to_change = random.choice(list(CleaningPolicy))
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:68]: exclude current cleaning policy from choices

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 31]
```diff
@@ -0,0 +1,129 @@
...
+            cache.set_cleaning_policy(policy_to_change)
+            if cache.get_cleaning_policy() != policy_to_change:
+                TestRun.fail("Wrong cleaning policy after chane it during I/O")
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:129]: as above

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 32]
```diff
@@ -28,20 +35,133 @@
...
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_incremental_load_hidden_core():
+    """
+    title: Incremental load test with hidden core
+    description: |
+        Test incremental load and core pool functionality with hidden core partition
+    pass_criteria:
+      - cores after adding to core pool are in detached state
+      - visible cores after start and load should be in active state
+      - hidden core after load should be in detached state
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:108]:
```suggestion
      - core with missing device should be in detached state after load
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 33]
```diff
@@ -28,20 +35,133 @@
...
+@pytest.mark.os_dependent
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_incremental_load_hidden_core():
+    """
+    title: Incremental load test with hidden core
+    description: |
+        Test incremental load and core pool functionality with hidden core partition
+    pass_criteria:
+      - cores after adding to core pool are in detached state
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:106]:
```suggestion
      - cores are in detached state after adding to core pool
```
or
```suggestion
      - after adding to core pool cores are in detached state
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 34]
```diff
@@ -28,20 +35,133 @@
...
+      - visible cores after start and load should be in active state
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:107]:
```suggestion
      - visible cores should be in active state after start and load
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 35]
```diff
@@ -28,20 +35,133 @@
...
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(2, Unit.GibiByte)])
+        core_device.create_partitions([Size(4, Unit.GibiByte)] * 3)
+
+        cache_dev = cache_device.partitions[0]
+        core_dev_list = core_device.partitions
+
+    with TestRun.step("Start cache and add core device"):
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:121]:
```suggestion
    with TestRun.step("Start cache and add cores"):
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 36]
```diff
@@ -28,20 +35,133 @@
...
+        cache = casadm.start_cache(cache_dev=cache_dev, force=True)
+        core_list = [cache.add_core(core_dev=core_dev) for core_dev in core_dev_list]
+
+    with TestRun.step("Check if all cores has active status"):
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:125]:
```suggestion
    with TestRun.step("Check if all cores have active status"):
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 37]
```diff
@@ -28,20 +35,133 @@
...
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_incremental_load_basic():
+    """
+    title: Incremental load test basic
+    description: |
+        Test incremental load and core pool functionality
+    pass_criteria:
+      - cores after start and load should be in active state and cache in running state
+      - cores after adding to core pool are in inactive state and cache in incomplete state
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:49]:
```suggestion
      - after start and load cores should be in active state and cache in running state
      - after adding to core pool cores are in inactive state and cache in incomplete state
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 38]
```diff
@@ -28,20 +35,133 @@
...
+        core_device.create_partitions([Size(4, Unit.GibiByte)] * core_count)
+
+        cache_dev = cache_device.partitions[0]
+        core_dev_list = core_device.partitions
+
+    with TestRun.step("Start cache and add core device"):
+        cache = casadm.start_cache(cache_dev=cache_dev, force=True)
+        core_list = [cache.add_core(core_dev=core_dev) for core_dev in core_dev_list]
+
+    with TestRun.step("Check if all cores has active status"):
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:67]:
```suggestion
    with TestRun.step("Check if all cores have active status"):
```

File: test/functional/tests/initialize/test_recovery.py [hunk 39]
```diff
@@ -0,0 +1,196 @@
...
+
+    with TestRun.step("Mount exported object"):
+        core.mount(mount_point)
+        if not check_if_file_exists(core.mount_point):
+            TestRun.LOGGER.error(f"Mounting core device {mount_point} failed")
+
+    with TestRun.step("Calculate test file md5sums after recovery"):
+        core_mnt_md5s_after = File(f"{core.mount_point}/test").md5sum()
+
+    with TestRun.step("Compare test file md5 sums for cores and core devices"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:191]:
```suggestion
    with TestRun.step("Compare test file md5 sums"):
```

File: test/functional/tests/initialize/test_recovery.py [hunk 40]
```diff
@@ -0,0 +1,196 @@
...
+        if core_mnt_md5s_before != core_mnt_md5s_after:
+            TestRun.fail(
+                f"MD5 sums of core before and after does not match"
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:194]:
```suggestion
                f"MD5 sums do not match\n"
```

File: test/functional/tests/initialize/test_recovery.py [hunk 41]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (Deixx) [inline test/functional/tests/initialize/test_recovery.py:110]:
```suggestion
                f"MD5 sums do not match\n"
```

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 42]
```diff
@@ -0,0 +1,137 @@
...
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_change_cleaning_policy_during_io_raw():
+    """
+    title: Test for changing the cleaning policy during background IO.
+    description: |
+        Stress test to change the cleaning policy during background IO operations on CAS raw
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:30]:
```suggestion
        Stress test to change the cleaning policy during background IO operations on raw
```

File: test/functional/tests/stress/test_stress_cleaning_policy_change.py [hunk 43]
```diff
@@ -0,0 +1,137 @@
...
+        exported object with a Write-Back cache mode.
```
Comment (Deixx) [inline test/functional/tests/stress/test_stress_cleaning_policy_change.py:31]:
```suggestion
        exported object in Write-Back cache mode.
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 44]
```diff
@@ -28,20 +35,134 @@
...
+    with TestRun.step("Start cache and add cores"):
+        cache = casadm.start_cache(cache_dev=cache_dev, force=True)
+        core_list = [cache.add_core(core_dev=core_dev) for core_dev in core_dev_list]
+
+    with TestRun.step("Check if all cores have active status"):
+        for core in core_list:
+            if core.get_status() is not CoreStatus.active:
+                TestRun.fail(f"Core {core.core_id} should be active but is {core.get_status()}.")
+
+        missing_core = cache.get_core_devices()[-1]
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:131]:
```suggestion
        core_with_missing_device = cache.get_core_devices()[-1]
```

File: test/functional/tests/incremental_load/test_incremental_load.py [hunk 45]
```diff
@@ -28,20 +35,134 @@
...
+        cache = casadm.load_cache(cache_dev)
+        if cache.get_status() is not CacheStatus.incomplete:
+            TestRun.fail(
+                f"Cache {cache.cache_id} should be incomplete but is " f"{cache.get_status()}."
+            )
+        for core in cache.get_core_devices():
+            if core.get_status() is not CoreStatus.active:
+                TestRun.fail(f"Core {core.core_id} should be Active but is {core.get_status()}.")
+        if missing_core.get_status() is not CoreStatus.inactive:
+            TestRun.fail(f"Remove core should be Inactive but is {missing_core.get_status()}.")
```
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:149]:
```suggestion
            TestRun.fail(f"Core without backend should be Inactive but is {missing_core.get_status()}.")
```
or `Core with missing device`

File: test/functional/tests/incremental_load/test_incremental_load.py (hunk 44 above)
Comment (Deixx) [inline test/functional/tests/incremental_load/test_incremental_load.py:131]: off-comment: in Cache class get_core_devices() should be renamed to get_cores()

File: test/functional/tests/initialize/test_recovery.py (hunk 40 above)
Comment (Deixx) [reply test/functional/tests/initialize/test_recovery.py]: add line break

File: test/functional/tests/initialize/test_recovery.py (hunk 41 above)
Comment (Deixx) [reply test/functional/tests/initialize/test_recovery.py]: add line break

File: test/functional/tests/initialize/test_simulation_startup.py [hunk 46]
```diff
@@ -0,0 +1,90 @@
...
+        core = cache.get_core_devices()[0]
+        if core.get_status() is not CoreStatus.active:
+            TestRun.fail(
+                f"Core {core.core_id} should be active but is in {core.get_status()} " f"state."
+            )
+
+    with TestRun.step("Stop cache instance using casctl"):
+        TestRun.executor.run_expect_success(ctl_stop())
+
+    with TestRun.step("Triggering udev"):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_simulation_startup.py:74]:
```suggestion
    with TestRun.step("Trigger udev"):
```

**Review by Deixx — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**